pandas/tests/apply/test_frame_apply.py

```diff
@@ -1551,3 +1551,11 @@ def foo(x):
     df = DataFrame({"a": [1, 2, 3]})
     with tm.assert_produces_warning(UserWarning, match="Hello, World!"):
         df.agg([foo])
+
+
+def test_apply_return_type():
+    # GH 35517
+    df = DataFrame([["foo"]])
+    result = type(df.apply(lambda col: np.array("bar")).iloc[0])
+    expected = np.ndarray
+    assert result == expected
```

Comment:
can you actually construct the expected exactly and use tm.assert_equal

Reply:
Hi @jreback 
I think I cannot do it using tm.assert_equal. The expected is `np.ndarray` type, not `DataFrame` type.

Reply:
Could you assert the full result of `df.apply(lambda col: np.array("bar"))`? Not including the the `iloc[0]` part or `type`

Reply:
Yes. I have remove iloc[0] part. Using `tm.assert_series_equal` in test case.